File: public/tools/elements/dashboard/GtDashboard.ts
import {View} from "../../client/router/View";
import {PolymerElement} from "../../polymer/PolymerElement";
import {Inject, Element} from "../../polymer/Annotations";
import {Server} from "../../client/server/Server";
import {DashboardNews} from "./DashboardNews";
import {DashboardShoutbox} from "./DashboardShoutbox";
import {DashboardOnlines} from "./DashboardOnlines";
import {ProfileUser} from "../profile/ProfileUser";

@View("dashboard", () => [{title: "Dashboard", link: "/dashboard", active: true}])
@Element({
	selector: "gt-dashboard",
	template: "/assets/views/dashboard.html",
	dependencies: [DashboardNews, DashboardShoutbox, DashboardOnlines, ProfileUser]
})
export class GtDashboard extends PolymerElement {
	@Inject private server: Server;
}

File: client/src/gt/components/dashboard/GtDashboard.scala
package gt.components.dashboard

import gt.components.profile.ProfileUser
import gt.components.widget.GtBox
import gt.components.{GtHandler, Tab, View}
import utils.jsannotation.js
import xuen.Component

object GtDashboard extends Component[GtDashboard](
	selector = "gt-dashboard",
	templateUrl = "/assets/imports/views/dashboard.html",
	dependencies = Seq(GtBox, DashboardNews, DashboardShoutbox, DashboardOnlines, ProfileUser)
) with View {
	val module = "dashboard"
	val tabs: TabGenerator = (selector, path, user) => Seq(
		Tab("Dashboard", "/dashboard", active = true)
	)
}

@js class GtDashboard extends GtHandler {

}

